posix/posix.c

```diff
@@ -1147,10 +1147,10 @@ int posix_ftruncate(int fildes, off_t length)
 
 	err = posix_getOpenFile(fildes, &f);
 	if (err == 0) {
-		err = posix_truncate(&f->oid, length);
+		err = ((f->status & O_RDONLY) != 0) ? -EBADF : posix_truncate(&f->oid, length);
```

Comment:
I'd rather use `if` here as one of the clauses has side effect

Reply:
It's a valid argument, on the other hand it seems wasteful to me to do
if (a) var = b;
else if (b) var = c;